### PR TITLE
Use `XDG_DATA_HOME` for install prefix in init script when set

### LINKED ISSUE #136: Use XDG_DATA_HOME environment variable as default_prefix instead of hardcoded ~/.local/share
Title. By default, `XDG_DATA_HOME` is set to ~/.local/share, but allows users to change their system's default location (for example, I have it as `~/bin`). As far as I am aware, macOS and FreeBSD supports setting these variables as well. I am willing to PR to add this if permitted :)

### PR DESCRIPTION
**Pull request by AlexFischbacher**
Closes #136.

### CODE REVIEW

**Comment by Theodus** [comment]
CI failure: `sh: XDG_DATA_HOME: parameter not set`

File: ponyup-init.sh [hunk 1]
```diff
@@ -3,7 +3,12 @@
 set -o errexit
 set -o nounset
 
-default_prefix="$HOME/.local/share"
+if [ -z "$XDG_DATA_HOME" ]; then
```
Comment (Theodus) [inline ponyup-init.sh:6]: This should be changed to `-z "${XDG_DATA_HOME+x}"` to handle the case that the environment variable is unset.

File: ponyup-init.sh [hunk 2]
```diff
@@ -3,7 +3,12 @@
...
+    default_prefix="$HOME/.local/share"
```
Comment (Theodus) [inline ponyup-init.sh:7]: The indentation is off here. The rest of the file uses 2 spaces.

File: ponyup-init.sh (hunk 1 above)
Comment (AlexFischbacher) [reply ponyup-init.sh]: Fixed.

File: ponyup-init.sh (hunk 2 above)
Comment (AlexFischbacher) [reply ponyup-init.sh]: Fixed, forgot to put it here though (whoops).

**Comment by SeanTAllen** [comment]
@Theodus don't forget to add a change log for this

**Pull request merged**